vocab.py

```diff
@@ -20,9 +28,92 @@ class ValidDataSourceName(Enum):
     Each data source name should be defined in ValidDataSourceName.
     NAME_OF_DATA_SOURCE = auto()
     """
-    
+
     CANVAS_API = auto()
     CANVAS_ZOOM_MEETINGS = auto()
     KALTURA_API = auto()
     UNIZIN_DATA_PLATFORM_EVENTS = auto()
     UNIZIN_DATA_WAREHOUSE = auto()
+
+
+class DataSourceStatus:
+    '''
+    Each data source process must return an instance of this class, which tells the name of the
+    data source and the time at which data from it was last updated.
+    '''
+    def __init__(
+            self,
+            dataSourceName: ValidDataSourceName = NotImplemented,
```

Comment:
PEP-8 camelCase is [really only *allowed* to conform](https://david.goodger.org/projects/pycon/2007/idiomatic/handout.html#naming) to pre-existing conventions. Is there some reason you're preferring this here in instance variables but joined lower in other places?? 

Generally these conventions were for code that was converted over from another language.

Reply:
I believe I am consistent in using camelCase.  I don't know of any place I use `joined_lower` here.  If I did, please point it out.  I'll check for it again.

I use `JOINED_CAPS` when naming constants.

I've discussed this before with Sam.  He wasn't very enthusiastic about my preference, but he thought he could live with it if it's confined to specific module(s).

Reply:
@lsloan, since this is a shared module that everyone will have to use and the prevailing style of this project is `snake_case`, I would prefer that you use that in this file, please.

Reply:
Yeah, I was going to basically say the same as that's the Python style. Unless this code was inspired from a module written in Java or if this was already the convention of the project it should be joined_lower. This is a shared module that others will contribute to.

**Naming**
* joined_lower for functions, methods, attributes
* joined_lower or ALL_CAPS for constants
* StudlyCaps for classes
* camelCase only to conform to pre-existing conventions
* Attributes: interface, _internal, __private

Reply:
I have a **_rant_** about PEP 8, but I'll not waste time with that here.

Reply:
I should say that I looked over these reviews this morning before 09:00.  I didn't look back until after @ssciolla mentioned in a meeting today that he specifically asked me to not use camelCase.  I didn't see his message when I committed some changes and prompted him and @jonespm to review again.  I wasn't purposely ignoring his request.  I guess I need to spend more time checking for GitHub comments in the GH UI and my email before committing changes and requesting reviews.

Reply:
No worries, sorry if I came on a little strong, totally reasonable that you missed the comment. I just figured I'd have a direct conversation with you and hear what your thinking was.